core/src/main/kotlin/org/opensearch/alerting/core/model/DataSources.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.core.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.ToXContentObject
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+
+data class DataSources(
+    val queryIndex: String? = null,
```

Comment:
Can this be a list too? Or we are assuming it will always be one index to begin with?
Also, for other monitors such as bucket level holds the OS query directly. We will need to think about them as well.

Reply:
Let's begin with single query index to which all queries of a monitor are written. If different queries are to be written to different indices, users can always different monitors with the different custom query indices.